src/main/java/redis/clients/jedis/JedisSlotBasedConnectionHandler.java

```diff
@@ -44,9 +43,12 @@ public Jedis getConnection() {
         if (result.equalsIgnoreCase("pong")) return jedis;
 
         pool.returnBrokenResource(jedis);
-      } catch (JedisConnectionException ex) {
+      } catch (JedisException ex) {
         if (jedis != null) {
-          pool.returnBrokenResource(jedis);
+          try {
```

Comment:
OK, let's remove that nested try-catch if needed.
I am not so familiar with that code to tell whether replacing old call to returnBrokenResource() with call to close() is good or bad, and why.
I can say no pros or cons for these two suggested changes, but replacing try-catch with try-finally seems to be a very bad idea: earlier that code catched JedisConnectionException from ping(), now it shall catch any JedisException as well, but you suggest to let that exception be thrown further up instead of catching.